spec/index.adoc

```diff
@@ -626,64 +627,97 @@ See <<state-tree>> for details on the state tree.
-* `method_name` (`text`): Name of the canister method to call
-* `arg` (`blob`): Argument to pass to the canister method
-
-The HTTP response to a query call contains a list of signatures for the returned response produced by the individual IC nodes that computed the same returned response.
-Every such signature (whose type is denoted as `node-signature-of-query-response`) is a CBOR (see <<cbor>>) map with the following fields:
-
-* `timestamp` (`nat`): the timestamp of the signature.
-* `signature` (`blob`): the actual signature.
-* `identity` (`principal`): the principal of the node producing the signature.
+* `method_name` (`text`): Name of the canister method to call.
+* `arg` (`blob`): Argument to pass to the canister method.
 
-If the call resulted in a reply, the response is a CBOR (see <<cbor>>) map with the following fields:
+If the query call resulted in a reply, the response is a CBOR (see <<cbor>>) map with the following fields:
 
 * `status` (`text`): `"replied"`
-* `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data `<R>`.
-* `node_signatures` (`[+ node-signature-of-query-response]`): a list of node signatures for the returned query response
-  computed for the following value
-+
-....
-hash_of_map({
-  status: "replied",
-  reply: {arg: <R>},
-  timestamp: <t>,
-  request_id: hash_of_map(content)
-})
-....
-+
-where `hash_of_map` is the <<hash-of-map, representation-independent hash>>, `content` is the CBOR map from the request, and `<t>` is the timestamp of the node when the signature was produced.
+* `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data.
 
 If the call resulted in a reject, the response is a CBOR map with the following fields:
 
 * `status` (`text`): `"rejected"`
 * `reject_code` (`nat`): The reject code (see <<reject-codes>>).
 * `reject_message` (`text`): a textual diagnostic message.
 * `error_code` (`text`): an optional implementation-specific textual error code (see <<error-codes>>).
-* `node_signatures` (`[+ node-signature-of-query-response]`): a list of node signatures for the returned query response
-  computed for the following value
-+
+
+[#http-signed-query]
+=== Request: Signed query call
+
+NOTE: The signed query call API is considered EXPERIMENTAL. Canister developers must be aware that the API may evolve in a non-backward-compatible way.
+
+A *signed* query call is a query call amended with signatures produced by IC nodes evaluating the query call to certify the query response.
+
+In order to make a signed query call to a canister, the user makes a POST request to `/api/v2/canister/<effective_canister_id>/signed_query`.
+The request body has the same structure as the request body for <<http-query>> with the `content` map having the following *additional* fields:
+
+* (optional) `return_signatures` (`text`): Determines whether and how many signatures are returned in the response; possible values are `"no"`, `"one"`.
```

Comment:
Shall we go for `none`, instead of `no`? Or is there something more suitable, e.g., something similar to an enum, to encode the choices here?

Reply:
> Shall we go for none, instead of no?

Sure, why not.

> Or is there something more suitable, e.g., something similar to an enum, to encode the choices here?

There's no native support for enum in CBOR. So we need to use integers or text. I went for text as integers could suggest we control how many signatures we precisely get.